src/main/java/uk/gov/ons/census/casesvc/service/CaseReceiptService.java

```diff
@@ -17,51 +24,121 @@
   private final CaseService caseService;
   private final CaseRepository caseRepository;
 
+  private Map<Key, Rule> rules = new HashMap<>();
+
   public CaseReceiptService(CaseService caseService, CaseRepository caseRepository) {
     this.caseService = caseService;
     this.caseRepository = caseRepository;
+    setUpRules();
+  }
+
+  private void setUpRules() {
+    /*
+     This table is based on: https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?spaceKey=SDC&title=Receipting
+     The invalid rows are missing from here and are handled collectively by throwing a " does not map to any valid processing rule" RunTimeException
+    */
+    rules.put(new Key("HH", "U", "HH"), new Rule(receiptCase, true));
+    rules.put(new Key("HH", "U", "Cont"), new Rule(noActionRequired, false));
+    rules.put(new Key("HI", "U", "Ind"), new Rule(receiptCase, true));
+    rules.put(new Key("CE", "E", "CE1"), new Rule(receiptCase, true));
+    rules.put(new Key("CE", "U", "Ind"), new Rule(incrementAndReceipt, true));
+    rules.put(new Key("SPG", "E", "HH"), new Rule(noActionRequired, false));
+    rules.put(new Key("SPG", "E", "Ind"), new Rule(noActionRequired, false));
+    rules.put(new Key("CE", "E", "Ind"), new Rule(incremenNoReceipt, true));
+    rules.put(new Key("SPG", "U", "HH"), new Rule(receiptCase, true));
+    rules.put(new Key("SPG", "U", "Ind"), new Rule(noActionRequired, false));
+    rules.put(new Key("SPG", "U", "Cont"), new Rule(noActionRequired, false));
   }
 
   public void receiptCase(UacQidLink uacQidLink, EventTypeDTO causeEventType) {
     Case caze = uacQidLink.getCaze();
 
     if (caze.isReceiptReceived()) return;
 
-    if (iscontinuationQuestionnaireTypes(uacQidLink.getQid())) return;
+    Key ruleKey = makeRulesKey(caze, uacQidLink);
 
-    if (caze.getCaseType().equals("CE") && isIndividualQuestionnaireType(uacQidLink.getQid())) {
-      incrementActualResponseAndSetReceiptedIfAppropriate(caze, causeEventType);
-      return;
+    if (!rules.containsKey(ruleKey)) {
+      throw new RuntimeException(ruleKey.toString() + " does not map to any valid processing rule");
     }
 
-    caze.setReceiptReceived(true);
-    caseService.saveCaseAndEmitCaseUpdatedEvent(
-        caze, buildMetadata(causeEventType, ActionInstructionType.CLOSE));
+    Rule rule = rules.get(ruleKey);
+    Case lockedCase = rule.run(caze);
+
+    if (rule.saveAndEmitCase) {
```

Comment:
No getter?

Reply:
I'll stick one on, not an expert on the right style for using private local classes.

Reply:
done